Feature/fix tests

eran-gil opened the pull request:
As the integration tests didn't work I worked to fix them. Please review so the local tests will work in the future with elasticsearch's latest versions.

eran-gil committed:
removed bad field from template to avoid errors when logging to elastic

eran-gil committed:
Elastic 2.0 and above does not allow '.' in fields, so fixed it in the flatten json logic to replace the '.' with '_'

eran-gil committed:
Since the ttl field is deprecated in 2.0 and removed in 5.0, I removed the ttl test as it fails when logging to elastic 2.0 and above.

urielha commented on src/log4stash/Filters/JsonFilter.cs:
```diff
@@ -76,7 +76,7 @@ private static void ScanToken(IDictionary<string, object> logEvent, JToken token
         private static string Join(string prefix, string name)
         {
-            return (string.IsNullOrEmpty(prefix) ? name : prefix + "." + name);
+            return (string.IsNullOrEmpty(prefix) ? name : prefix + "_" + name);
```
Is the dot scrambling the ES?
From which version?

urielha replied:
Can you extract "_" into a property? so it could be configured from outside?
Thanks :)

eran-gil replied:
The dot is not allowed in field names since elastic 2.0...
https://www.elastic.co/blog/introducing-the-de_dot-filter
"One of these changes was that Elasticsearch 2.0 does not support field names with a . (or dot character) in them"

eran-gil replied:
And done :)

eran-gil committed:
Merge branch 'master' into feature/fix_tests

urielha reviewed (approved):
Please have a look at my comment

eran-gil committed:
The separator in the json filter (used when flattening jsons) is now configurable through the appnder's configuration.
Also updated the documentation and tests to support the new property in the filter.

urielha commented:
Thanks :)
